lib/ui/survey_screen/cubit/survey_cubit.dart

```diff
@@ -10,75 +10,61 @@ class SurveyCubit extends Cubit<SurveyState> {
-    'Какие типы баз данных бывают',
-    'Какие есть методы коммуникации с Rest API',
-    'Что такое билд контекст и зачем он нужен',
-    'Какие протоколы для обмена данными ты знаешь?',
-    'Что такое DAO паттерн?',
-    'Что такое Repository pattern?',
-    'Какие инструменты Flutter вы используете для анализа и оптимизации производительности?',
-    'Объясни принцип работы JIT и AOT',
-    'Какие базовые типы вы знаете? Какие структуры данных вам известны?',
-    'В чем разница между dynamic и Object?',
-    'В чем разница между final, var и const?',
-    'Могу ли я вызвать метод add() для константного списка?',
-    'От чего наследуются типы?',
-    'В чём отличие List и Set?',
-    'Можем ли мы изменить объект, который final?',
-    'Что такое typedef и когда его можно использовать?',
-    'Какие методы объектов вы знаете?',
-    'Есть ли в Dart интерфейсы?',
-    'Что такое абстрактный класс? Можем ли создать экземпляр абстрактного класса?',
-    'Зачем использовать геттеры и сеттеры?'
-  ];
-
-
-  void uploadNewQuestion() async {
+  Future<void> uploadNewQuestion() async {
     emit(SurveyLoading());
-    final randomQuestion = _getRandomQuestion();
-    emit(SurveySuccessLoaded(randomQuestion));
+    final question = await _getRandomQuestionFromFirestore();
+    if (question != null) {
+      emit(SurveySuccessLoaded(question));
+    } else {
+      emit(SurveyInitial());
+    }
   }
 
-  void registerFail(String question) async {
+  Future<void> registerFail(String question) async {
     emit(SurveyLoading());
     await repository.updateFail(question);
-    final randomQuestion = _getRandomQuestion();
-    emit(SurveySuccessLoaded(randomQuestion));
+    final newQuestion = await _getRandomQuestionFromFirestore();
+    if (newQuestion != null) {
+      emit(SurveySuccessLoaded(newQuestion));
+    } else {
+      emit(SurveyInitial());
+    }
   }
 
-  void getQuestion() async {
+  Future<void> getQuestion() async {
     emit(SurveyLoading());
-    final randomQuestion = _getRandomQuestion();
-    emit(SurveySuccessLoaded(randomQuestion));
+    final question = await _getRandomQuestionFromFirestore();
+    if (question != null) {
+      emit(SurveySuccessLoaded(question));
+    } else {
+      emit(SurveyInitial());
+    }
   }
 
-  String _getRandomQuestion() {
-    final random = Random();
-    return questions[random.nextInt(questions.length)];
+  Future<String?> _getRandomQuestionFromFirestore() async {
+    final questions = await repository.getQuestions();
+    final unansweredQuestions =
+        questions.where((q) => q['answered_correctly'] == false).toList();
+    if (unansweredQuestions.isNotEmpty) {
+      final random = Random();
+      final question =
+          unansweredQuestions[random.nextInt(unansweredQuestions.length)];
+      return question['text'] as String?;
+    } else {
+      emit(SurveyInitial());
+    }
+    return null;
+  }
+
+  Future<void> registerCorrectAnswer(String questionText) async {
```

Comment:
create a dto that will be for the question, which will store a string with the question and a boolean, pass it and parse